### DISCUSSION TITLE
Metal Resource Residency With Bindless

### DISCUSSION THREAD

**Discussion by JonahGoldsmith** [KhronosGroup/MoltenVK#2140] [Q&A] [answered]
When using bindless model how does MoltenVK track resources that need to be resident? If i just have a huge array of UAV resources and im only calling BindDescriptorSet once what goes on behind the scenes? Is every draw call hit with a useResource call for every single resource created as a UAV? Or is there behind the scenes tracking somehow

**Comment by billhollings** [comment, marked as answer]
Yes. Currently MoltenVK must make `useResource` calls for each resource in bound descriptor sets.

We're looking for ways to optimize that, but it is what it is for Metal.
